### PR TITLE
All Neighbors System Dashboard - Returns tab

### PR DESCRIPTION
**Pull request by eanders**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Provides:
* Returns tab to the All Neighbors System Dashboard

## Type of change
[//]: # 'remove options that are not relevant'

- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: app/javascript/all_neighbors_system_dashboard_stack.js [hunk 1]
```diff
@@ -20,19 +20,24 @@ class AllNeighborsSystemDashboardStack {
...
   }
 
   test() {
     console.debug(this)
   }
 
+  getActiveConfig() {
+    return {}
+  }
+
```
Comment (dtgreiner) [inline app/javascript/all_neighbors_system_dashboard_stack.js:34]: do we need an empty hash version of getActiveSeries() also?

File: drivers/all_neighbors_system_dashboard/app/models/all_neighbors_system_dashboard/returns_to_homelessness.rb [hunk 2]
```diff
@@ -8,198 +8,172 @@ module AllNeighborsSystemDashboard
...
-      where_clause = date_query(range)
-      scope.where(where_clause)
+    def project_types_with_data
+      # FIXME: needs to be based on the data available
+      project_types
+      # raise 'hi'
+      # @project_types_with_data ||= line_data[:project_types].
+      #   # Reject any project types where we have NO data
+      #   reject { |m| m[:count_levels].flatten.map { |n| n[:monthly_counts] }.flatten(2).map(&:last)&.all?(0) }.
+      #   map { |m| m[:project_type] }
```
Comment (dtgreiner) [inline drivers/all_neighbors_system_dashboard/app/models/all_neighbors_system_dashboard/returns_to_homelessness.rb]: Is this still in progress?

**Review by dtgreiner — APPROVED** [review]

File: drivers/all_neighbors_system_dashboard/app/models/all_neighbors_system_dashboard/returns_to_homelessness.rb (hunk 2 above)
Comment (eanders) [reply drivers/all_neighbors_system_dashboard/app/models/all_neighbors_system_dashboard/returns_to_homelessness.rb]: Yes!  Thank you.  Fix pushed.

File: app/javascript/all_neighbors_system_dashboard_stack.js (hunk 1 above)
Comment (eanders) [reply app/javascript/all_neighbors_system_dashboard_stack.js:34]: Probably a good idea!  Thanks.

File: app/javascript/all_neighbors_system_dashboard_stack.js [hunk 3]
```diff
@@ -20,19 +20,28 @@ class AllNeighborsSystemDashboardStack {
...
+  getActiveSeries() {
+    return {};
+  }
+
   inDateRange(dateString, range) {
     const [year, month, day] = dateString.split('-')
     //ruby date month is 1 based while js date month is 0
     const date = Date.parse(new Date(year, month-1, day))
     const [s, e] = range
+    // console.log('comparing dates', new Date(date), new Date(s), new Date(e))
```
Comment (dtgreiner) [inline app/javascript/all_neighbors_system_dashboard_stack.js]: Are we wanting to keep the testing comments in the code?

File: app/javascript/all_neighbors_system_dashboard_stack.js [hunk 4]
```diff
@@ -477,40 +485,204 @@ class AllNeighborsSystemDashboardRTHStack extends AllNeighborsSystemDashboardSta
...
+    const monthly_counts = categories.map((_, cat_index) => {
+      // Gather up all of the returns or placements for the given category
+      return d3.sum(monthly_counts_for_name.map((counts) => counts[cat_index]))
+    })
+    return col.concat(monthly_counts)
+  }
+
   getConfig() {
-    const fitLabels = this.fitLabels
+    // const fitLabels = this.fitLabels
```
Comment (dtgreiner) [inline app/javascript/all_neighbors_system_dashboard_stack.js]: I'm assuming this is staying in to be possibly turned on later

**Review by dtgreiner — APPROVED** [review]

File: app/javascript/all_neighbors_system_dashboard_stack.js (hunk 3 above)
Comment (eanders) [reply app/javascript/all_neighbors_system_dashboard_stack.js]: Nope, good call.  Thanks.

File: app/javascript/all_neighbors_system_dashboard_stack.js (hunk 4 above)
Comment (eanders) [reply app/javascript/all_neighbors_system_dashboard_stack.js]: Yes, but I think it's safe to remove now.  Thanks.

**Pull request merged**